### PR TITLE
feat(notification) : update a notification

### PR DESCRIPTION
**Pull request by Med1233**
## Description

**Fixes** # https://centreon.atlassian.net/browse/MON-19625

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by bdauria — APPROVED** [review]

File: centreon/doc/API/v23.10/Configuration/Notification/Notification.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    '403':
+      $ref: '../../Common/Response/Forbidden.yaml'
+    '404':
+      $ref: '../../Common/Response/NotFound.yaml'
+    '500':
+      $ref: '../../Common/Response/InternalServerError.yaml'
+patch:
+  tags:
+    - Notification
+  summary: "Partial update a notification definition"
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Notification/Notification.yaml:50]:
```suggestion
  summary: "Partial update of a notification definition"
```

File: centreon/doc/API/v23.10/Configuration/Notification/Schema/Notification.Details.yaml [hunk 2]
```diff
@@ -0,0 +1,84 @@
+type: object
+properties:
+  id:
+    type: integer
+    description: "ID of the notification"
+    example: 1
+  name:
+    type: string
+    description: "Name of notification"
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Notification/Schema/Notification.Details.yaml:9]:
```suggestion
    description: "Name of the notification"
```

File: centreon/doc/API/v23.10/Configuration/Notification/Schema/Notification.Details.yaml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      properties:
+        channel:
+          type: string
+          description: "Channel used to send the notification"
+        subject:
+          type: string
+          description: "Subject of the message"
+        message:
+          type: string
+          description: "content of the message"
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Notification/Schema/Notification.Details.yaml:39]:
```suggestion
          description: "Content of the message"
```

File: centreon/doc/API/v23.10/Configuration/Notification/Schema/Notification.Details.yaml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    type: array
+    items:
+      type: object
+      properties:
+        id:
+          type: integer
+          description: "ID of the user"
+        name:
+          type: string
+          description: "name of the user"
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Notification/Schema/Notification.Details.yaml:50]:
```suggestion
          description: "Name of the user"
```

File: centreon/doc/API/v23.10/Configuration/Notification/Schema/Notification.Details.yaml [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+          items:
+            type: object
+            properties:
+              id:
+                type: number
+                description: "ID of the resource"
+                example: 1
+              name:
+                type: string
+                description: "name of the resource"
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Notification/Schema/Notification.Details.yaml:75]:
```suggestion
                description: "Name of the resource"
```

File: centreon/tests/api/features/Notification.feature [hunk 6]
```diff
@@ -487,6 +488,552 @@ Feature:
...
+                          "id": 2,
+                          "name": "service-grp2"
+                      }
+                  ]
+              }
+          ]
+      }
+    """
+
+  Scenario: Notification Update as non-admin
```
Comment (cg-tw) [inline centreon/tests/api/features/Notification.feature:909]:
```suggestion
  Scenario: Notification update as non-admin
```

File: centreon/tests/php/Core/Notification/Application/UseCase/UpdateNotification/UpdateNotificationTest.php [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+    $this->readAccessGroupRepository = $this->createMock(ReadAccessGroupRepositoryInterface::class);
+    $this->contactRepository = $this->createMock(ContactRepositoryInterface::class);
+    $this->resourceRepositoryProvider = $this->createMock(NotificationResourceRepositoryProviderInterface::class);
+    $this->resourceRepository = $this->createMock(NotificationResourceRepositoryInterface::class);
+    $this->dataStorageEngine = $this->createMock(DataStorageEngineInterface::class);
+    $this->presenterFormatter = $this->createMock(PresenterFormatterInterface::class);
+    $this->presenter = new UpdateNotificationPresenterStub($this->presenterFormatter);
+});
+
+it('should present an forbidden response when the user is not admin and does not have sufficient ACLs', function (): void {
```
Comment (cg-tw) [inline centreon/tests/php/Core/Notification/Application/UseCase/UpdateNotification/UpdateNotificationTest.php:62]:
```suggestion
it('should present a forbidden response when the user is not admin and does not have sufficient ACLs', function (): void {
```

File: centreon/tests/php/Core/Notification/Application/UseCase/UpdateNotification/UpdateNotificationTest.php [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+        $contact
+    ))($request, $this->presenter);
+
+    expect($this->presenter->responseStatus)
+        ->toBeInstanceOf(ForbiddenResponse::class)
+        ->and($this->presenter->responseStatus?->getMessage())
+        ->toBe(NotificationException::updateNotAllowed()->getMessage());
+});
+
+it('should present a not found response when the does not exists', function (): void {
```
Comment (cg-tw) [inline centreon/tests/php/Core/Notification/Application/UseCase/UpdateNotification/UpdateNotificationTest.php:81]: when the what?

File: centreon/tests/php/Core/Notification/Application/UseCase/UpdateNotification/UpdateNotificationTest.php [hunk 9]
```diff
@@ -0,0 +1,251 @@
...
+        $this->contactRepository,
+        $this->resourceRepositoryProvider,
+        $this->dataStorageEngine,
+        $contact
+    ))($request, $this->presenter);
+
+    expect($this->presenter->responseStatus)->toBeInstanceOf(NotFoundResponse::class);
+});
+
+it('should present an InvalidArgumentResponse when a different notification with same name exists', function (): void {
```
Comment (cg-tw) [inline centreon/tests/php/Core/Notification/Application/UseCase/UpdateNotification/UpdateNotificationTest.php:107]:
```suggestion
it('should present an InvalidArgumentResponse when a different notification with the same name exists', function (): void {
```

File: centreon/tests/php/Core/Notification/Application/UseCase/UpdateNotification/UpdateNotificationTest.php (hunk 8 above)
Comment (jeremyjaouen) [inline centreon/tests/php/Core/Notification/Application/UseCase/UpdateNotification/UpdateNotificationTest.php:81]:
```suggestion
it('should present a not found response when the notification does not exists', function (): void {
```

**Review by bdauria — APPROVED** [review]

**Review by dmyios — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Pull request merged**